Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -1004,24 +1011,63 @@ namespace AzToolsFramework
+            auto handlerName = AZ::Name(handlerId->m_name);
+            auto handlerPool = poolManager->GetPool<PropertyHandlerWidgetInterface>(handlerName);
+            if (!handlerPool)
+            {
+                auto* theDPE = GetDPE();
+                AZ_Assert(theDPE, "DPERowWidget should always be able to retrieve its DPE during widget creation!");
+                if (!theDPE)
+                {
+                    return nullptr;
+                }
+                AZStd::function<void(PropertyHandlerWidgetInterface&)> resetHandler = [](PropertyHandlerWidgetInterface& handler)
+                {
+                    DetachAndHide(handler.GetWidget());
+                };
+
+                AZStd::function<PropertyHandlerWidgetInterface*()> createHandler = [handlerId]()
+                {
+                    auto createdHandler = AZ::Interface<PropertyEditorToolsSystemInterface>::Get()->CreateHandlerInstance(handlerId);
+                    auto bareHandler = createdHandler.get();
+                    createdHandler.release();
+                    return bareHandler;
+                };
+
+                handlerPool = poolManager->CreatePool<PropertyHandlerWidgetInterface>(handlerName, resetHandler, createHandler)
+                                 .GetValue();
+                theDPE->RegisterHandlerPool(handlerPool);
+            }
+
             // store, then reference the unique_ptr that will manage the handler's lifetime
-            auto handler = AZ::Interface<PropertyEditorToolsSystemInterface>::Get()->CreateHandlerInstance(handlerId);
+            auto handler = handlerPool->GetInstance();
             handler->SetValueFromDom(domValue);
             createdWidget = handler->GetWidget();
             createdWidget->setEnabled(true);
-            m_widgetToPropertyHandlerInfo[createdWidget] = { handlerId, AZStd::move(handler) };
+            m_widgetToPropertyHandlerInfo[createdWidget] = { handlerId, AZStd::unique_ptr<PropertyHandlerWidgetInterface>(handler) };
         }
         return createdWidget;
     }
 
     void DPERowWidget::ReleaseHandler(HandlerInfo& handler)
     {
-        QTimer::singleShot(
-            0,
-            [interfacePointer = handler.hanlderInterface.get()]()
-            {
-                delete interfacePointer;
-            });
+        auto poolManager = static_cast<AZ::InstancePoolManager*>(AZ::Interface<AZ::InstancePoolManagerInterface>::Get());
+        auto handlerName = AZ::Name(handler.handlerId->m_name);
+        auto handlerPool = poolManager->GetPool<PropertyHandlerWidgetInterface>(handlerName);
+        if (handlerPool)
+        {
+            handlerPool->RecycleInstance(handler.hanlderInterface.get());
+        }
+        else
+        {
+            QTimer::singleShot(
+                0,
+                [interfacePointer = handler.hanlderInterface.get()]()
+                {
+                    delete interfacePointer;
+                });
```

Comment:
I was going to suggest deleteLater but alas this isn't a QObject, so I'm OK with this singleshot :)